cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -426,17 +426,16 @@ CacheAllocator<CacheTrait>::allocateInternalTier(TierId tid,
                                                  uint32_t expiryTime,
                                                  bool fromBgThread,
                                                  bool evict) {
-  util::LatencyTracker tracker{stats().allocateLatency_};
-
+  util::LatencyTracker tracker{stats().allocateLatency_, static_cast<size_t>(~fromBgThread)};
   SCOPE_FAIL { stats_.invalidAllocs.inc(); };
 
   // number of bytes required for this item
   const auto requiredSize = Item::getRequiredSize(key, size);
 
   // the allocation class in our memory allocator.
   const auto cid = allocator_[tid]->getAllocationClassId(pid, requiredSize);
-  util::RollingLatencyTracker rollTracker{
-      (*stats_.classAllocLatency)[tid][pid][cid]};
+
+  util::RollingLatencyTracker rollTracker{(*stats_.classAllocLatency)[tid][pid][cid]};
```

Comment:
I assume this will still track latency for both user and bg allocations? If yes, it would be nice to put some comment regarding that.

Reply:
We're not tracking rolling average latency for background threads separately. This feature is not in use anywhere. It was meant to provide a low overhead tracker for profile-driven batch evictions. Profile-driven approach has not been explored since Daniel Obiri worked on it a year back and found it didn't provide any significant improvement. I'm not sure Meta will accept the rolling stats upstream PR - it has not been reviewed yet.